scripts-foundry/Deployer.s.sol

```diff
@@ -0,0 +1,108 @@
+// SPDX-License-Identifier: Apache2.0
+pragma solidity 0.8.4;
+
+import {Script} from "forge-std/Script.sol";
+import {console} from "forge-std/console.sol";
+
+import {System} from "../contracts/System.sol";
+import {BtcLightClient} from "../contracts/BtcLightClient.sol";
+import {Burn} from "../contracts/Burn.sol";
+import {CandidateHub} from "../contracts/CandidateHub.sol";
+import {Foundation} from "../contracts/Foundation.sol";
+import {GovHub} from "../contracts/GovHub.sol";
+import {PledgeAgent} from "../contracts/PledgeAgent.sol";
+import {RelayerHub} from "../contracts/RelayerHub.sol";
+import {SlashIndicator} from "../contracts/SlashIndicator.sol";
+import {ValidatorSet} from "../contracts/ValidatorSet.sol";
+import {SystemRewardMock} from "../contracts/mock/SystemRewardMock.sol";
+
+
+contract Deployer is System, Script {
+
+    uint public constant ANVIL_CHAINID = 31337;
+    uint public constant GANACHE_CHAINID = 1337;
+
+    // @dev declared as state-varible to circumvent stack-too-deep error
+    Burn private burn; 
+    BtcLightClient private lightClient;
+    SlashIndicator private slashIndicator;
+    SystemRewardMock private systemReward;
+    CandidateHub private candidateHub;
+    PledgeAgent private pledgeAgent;
+    ValidatorSet private validatorSet;
+    RelayerHub private relayerHub;
+    Foundation private foundation;
+    GovHub private govHub;
+
+    function run() external {
+	    // vm.startBroadcast(); 
+        if (_isLocalTestnet()) {
+            _performActualDeployment();
+        } else {
+            // rely on the already deployed contracts
+        }
+        // vm.stopBroadcast();
+    }
+
+    function _performActualDeployment() private {        
```

Comment:
Deploy all platform-contracts and use updateContractAddr (we're in test mode) to complete with the necessary wiring